### PR TITLE
Fix semaphore tracking for CB simultaneously inflight

### PR DESCRIPTION
**Pull request by chrisforbes**
This series tidies up around CB submission a bit, and moves the semaphore sets into a new CB_SUBMISSION structure. This allows correct tracking of the different semaphore sets associated with each simultaneous submission of the same CB.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Chris, this looks great and I'll try to give it a run through our build server and do some more strenuous testing.  We can fix up the commit messages before they get pushed.

**Comment by mark-lunarg** [comment]
So far, have only seen that on Windows, running the layer validation tests results in:
`[ RUN ] VkLayerTest.CommandBufferTwoSubmits
Assertion failed: !pFence->submissions.size(), file C:\dev\xgl\Vulkan-LoaderAndValidationLayers\layers\core_validation.cpp, line 4352`

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -318,6 +318,22 @@ VkBufferViewCreateInfo *getBufferViewInfo(const layer_data *my_data, VkBufferVie
     return bv_it->second.get();
 }
```
Comment (mark-lunarg) [inline layers/core_validation.cpp]: These changes might also need to be run through clang-format.

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -2237,6 +2237,61 @@ TEST_F(VkLayerTest, ResetUnsignaledFence) {
...
 }
 
+TEST_F(VkLayerTest, CommandBufferSimultaneousUseSync)
+{
+    m_errorMonitor->ExpectSuccess();
+
+    ASSERT_NO_FATAL_FAILURE(InitState());
+    VkResult err;
+
+    // record (empty!) command buffer that can be submitted multiple times
```
Comment (mark-lunarg) [inline tests/layer_validation_tests.cpp]: Can these comments get capitalized?  That'd make them more consistent with the rest of the file.

**Comment by chrisforbes** [comment]
@mark-lunarg This has uncovered some more missing initialization -- will fix on Monday.

**Comment by chrisforbes** [comment]
@mark-lunarg fixed the test. In addition to the behavior it was supposed to exercise, `CommandBufferTwoSubmits` was resubmitting the /fence/  and blundering past the error for that. I've fixed it to be more precise.

**Comment by mark-lunarg** [comment]
Passes the tests now, thanks Chris.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**